src/LiveComponent/src/Resources/doc/index.rst

```diff
@@ -616,6 +617,36 @@ args but inject to your defined parameter with another name.::
         }
     }
 
+Actions and file uploads
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 2.2
+
+    The ability to pass arguments to actions was added in version 2.2.
+
+If you want live component to track and send files you first need
+to mark file upload inputs as ``file`` Stimulus target.::
+
+.. code-block:: twig
+
+    <input type="file" {{ stimulus_target('live', 'file') }} name="my_file" />
+
+Then, when defining action you need to use special ``files(name)`` modifier.::
+
+.. code-block:: twig
+
+    <div
+        data-action="change->live#action"
+        data-action-name="prevent|files(my_file)|upload"
+        <input type="file" {{ stimulus_target('live', 'file') }} name="my_file" />
+    </div>
+
+This will send files from ``my_file`` file input. When used without argument
+it would send all files from all ``file`` targets of the controller.
+
+If you want to send multiple files from a single input remember to suffix its' name
+with ``[]`` - both in HTML name attribute and ``files`` modifier argument.
+
```

Comment:
The missing piece (which I'm guessing you were waiting on to get more feedback) is the PHP logic. We're going to need a really nice documentation example of how the `LiveAction` looks, including executing validation (and re-rendering with an error), processing the file upload (i.e. moving it somewhere) and setting the final filename (e.g. `tulips.jpg`) from that process onto some other `LiveAction` property... or setting it into an entity and saving immediately.